src/sentry/migrations/0572_rm_cficache_symcache.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 3.2.20 on 2023-10-05 09:51
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+# These two filetypes have not been created ever since
+# <https://github.com/getsentry/sentry/commit/c14c8f0e3a30fb5c2221fdefa4c0f5167bdc8986#diff-08e824e030b14e7936ffeda41ccf81d8feed2cdcdfbc522d0c46d0064f3640c9L750>
+# But there seems to be quite some leftover `File` objects sticking around we want to remove.
+TYPES_TO_DELETE = ("project.symcache", "project.cficache")
+
+
+def rm_cficache_symcache(apps, schema_editor):
+    File = apps.get_model("sentry", "File")
+
+    # According to <https://develop.sentry.dev/database-migrations/#filters>:
+    # > it is better to iterate over the entire table instead of using a filter
+    # Also, there is no index on `type` anyway.
+    for file in RangeQuerySetWrapperWithProgressBar(File.objects.all()):
+        if file.type in TYPES_TO_DELETE:
+            file.delete()
```

Comment:
Because migrations use 'historical models', the logic in `File.delete()` won't actually run. Without that logic unreferenced blob records (and the actual blobs in storage) will be retained until `cleanup` tools run.

We haven't had much success in spawning celery jobs from migrations, but you could synchronously call `sentry.tasks.files.delete_unreferenced_blobs_region` which will remove blob records and their storage.

Reply:
> Because migrations use 'historical models', the logic in File.delete() won't actually run.

But will it actually correctly unlink the foreign keys in `FileBlobIndex` then?

As `delete_unreferenced_blobs_region` has an import inside of it, that does not suffer from the same problem then? Because the `FileBlob.delete` hook it actually responsible for removing the actual blobs from storage.

Reply:
> But will it actually correctly unlink the foreign keys in FileBlobIndex then?

Yes, Django will remove the junction table records through an ORM cascade.

> As delete_unreferenced_blobs_region has an import inside of it, that does not suffer from the same problem then? 

The imports in other functions do not suffer from the historical model issue. The historical models come from `apps.get_model()` in the migration. Django recommends that migrations don't use application code in them as application code can change and imports can break. Several times in the past we've ignored these recommendations and guarded against future failure by catching import errors that may occur in the future.

Reply:
I’m also perfectly fine with the `cleanup` job picking these orphan file blobs up at a later point, as long as they are being cleaned up eventually :-)
In that case I believe not calling into `delete_unreferenced_blobs_region` directly might speed up the migration itself a bit, and the blobs will be removed eventually, so things should be fine.